src/H5T.c

```diff
@@ -6638,3 +6639,43 @@ H5T__get_path_table_npaths(void)
 
     FUNC_LEAVE_NOAPI(ret_value)
 }
+
+/*-------------------------------------------------------------------------
+ * Function:    H5T_is_numeric_with_unusual_unused_bits
+ *
+ * Purpose:     Detect if a datatype is a numeric datatype (int, float, or
+ *              bitfield) with an unusual # of unused bits.  This means
+ *              that the precision (i.e. the # of bits used) is less than
+ *              the size of the datatype, at power-of-two boundaries.
+ *
+ * Return:      true/false on success, can't fail
+ *
+ *-------------------------------------------------------------------------
+ */
+bool
+H5T_is_numeric_with_unusual_unused_bits(const H5T_t *dt)
```

Comment:
Testing on PPC64LE by @hyoklee shows that this probably won't work correctly for the IBM long double type. Suggest either:

```C
#if LDBL_MANT_DIG == 106
    /* This won't work for the IBM long double type */
    HGOTO_DONE(false);
#else
    ...
#endif
```

or some other solution.